src/test/java/com/artipie/pypi/PypiITCase.java

```diff
@@ -23,205 +23,95 @@
-                "--index-url", this.url.string(ArtipieServer.BOB), "anypackage"
-            ),
-            new StringContains(
-                String.format(
-                    "403 Client Error: Forbidden for url: %spip/", this.url.string()
-                )
-            )
+        this.containers.assertExec(
+            "Failed to upgrade pip",
+            new MavenITCase.ContainerResultMatcher(),
+            "python", "-m", "pip", "install", "--upgrade", "pip"
         );
     }
 
     @Test
-    void pypiPublishShouldFailSinceForbidden() throws Exception {
-        this.init(false);
-        new TestResource("pypi-repo/example-pckg/dist/")
-            .addFilesTo(
-                this.storage,
-                new Key.From("pypi-repo", "example-pckg")
+    void installPythonPackage() throws IOException {
+        String meta = "pypi-repo/example-pckg/dist/artipietestpkg-0.0.3.tar.gz";
+        this.containers.putResourceToArtipie(
+            meta, "/var/artipie/data/my-python/artipietestpkg/artipietestpkg-0.0.3.tar.gz"
         );
-        this.cntn.execStdout("python3", "-m", "pip", "install", "--user", "--upgrade", "twine");
-        MatcherAssert.assertThat(
-            "Packages should not be uploaded",
-            this.cntn.execStdErr(
-                "python3", "-m", "twine", "upload", "--verbose",
-                "--repository-url", this.url.string(),
-                "-u", ArtipieServer.BOB.name(), "-p", ArtipieServer.BOB.password(),
-                "pypi-repo/example-pckg/*"
-            ),
-            new StringContains("HTTPError: 403 Forbidden")
+        meta = "pypi-repo/example-pckg/dist/artipietestpkg-0.0.3-py2-none-any.whl";
+        this.containers.putResourceToArtipie(
+            meta,
+            "/var/artipie/data/my-python/artipietestpkg/artipietestpkg-0.0.3-py2-none-any.whl"
         );
-        MatcherAssert.assertThat(
-            "Packages should not be saved in storage",
-            this.inStorage("artipietestpkg-0.0.3-py2-none-any.whl")
-                || this.inStorage("artipietestpkg-0.0.3.tar.gz"),
-            new IsEqual<>(false)
+        this.containers.assertExec(
+            "Failed to install package",
+            new MavenITCase.ContainerResultMatcher(Matchers.equalTo(0)),
+            "python", "-m", "pip", "install", "--trusted-host", "artipie", "--index-url",
+            "http://artipie:8080/my-python", "artipietestpkg"
         );
     }
 
-    @AfterEach
-    void tearDown() {
-        this.server.stop();
-        this.cntn.close();
-    }
-
-    private void init(final boolean anonymous) throws IOException {
-        this.storage = new FileStorage(this.tmp);
-        this.server = new ArtipieServer(
-            this.tmp, "my-pypi", this.config(anonymous)
+    @Test
+    void canUpload() throws Exception {
+        this.containers.assertExec(
+            "Failed to upload",
+            new MavenITCase.ContainerResultMatcher(Matchers.is(0)),
```

Comment:
@milkysoft let's add matcher for logs as it was in the old test:
```
new StringContainsInOrder(
                new ListOf<String>(
                    "Uploading artipietestpkg-0.0.3-py2-none-any.whl", "100%",
                    "Uploading artipietestpkg-0.0.3.tar.gz", "100%"
                )
)
```
it can be important as exec result will 0 in the case when directory to upload from is empty, so the test will false positivly pass.

Reply:
@olenagerasimova , stdout checks added with latest commit